src/Badge/Service/ClientStrategy.php

```diff
@@ -28,72 +28,65 @@ public function __construct(GithubClient $githubClient, BitbucketClient $bitbuck
         $this->bitbucketClient = $bitbucketClient;
     }
 
+    /**
+     * @throws SourceClientNotFound
+     * @throws \Http\Client\Exception
+     */
     public function getDefaultBranch(Repository $repository): string
     {
         $defaultBranch = '';
-        $source = $repository->getSource();
+
         $username = $repository->getUsername();
         $repositoryName = $repository->getName();
 
-        if (self::GITHUB_SOURCE !== $source && self::BITBUCKET_SOURCE !== $source) {
-            throw new SourceClientNotFound('Source Client '.$source.' not found');
+        if (!$repository->isSupported()) {
+            throw new SourceClientNotFound('Source Client '.$repository->getSource().' not found');
         }
 
-        switch ($source) {
-            case self::GITHUB_SOURCE:
-                /** @var Repo $repoApi */
-                $repoApi = $this->githubClient->api('repo');
-                $repoGitHubData = $repoApi->show($username, $repositoryName);
-                if (!$this->isValidGithubRepository($repoGitHubData)) {
-                    throw new RepositoryDataNotValid('Repository data not valid: '.(string) \json_encode($repoGitHubData));
-                }
-
-                $defaultBranch = $repoGitHubData['default_branch'];
-                break;
-
-            case self::BITBUCKET_SOURCE:
-                $repoBitbucketData = $this->bitbucketClient
-                    ->repositories()
-                    ->users($username)
-                    ->show($repositoryName);
-
-                if (!$this->isValidBitbucketRepository($repoBitbucketData)) {
-                    throw new RepositoryDataNotValid('Repository data not valid: '.(string) \json_encode($repoBitbucketData));
-                }
-
-                $defaultBranch = $repoBitbucketData['mainbranch']['name'];
-                break;
+        if ($repository->isGitHub()) {
+            /** @var Repo $repoApi */
+            $repoApi = $this->githubClient->api('repo');
+            $repoGitHubData = $repoApi->show($username, $repositoryName);
+            if (!$this->isValidGithubRepository($repoGitHubData)) {
+                throw new RepositoryDataNotValid('Repository data not valid: '.(string) \json_encode($repoGitHubData));
+            }
+
+            return (string) $repoGitHubData['default_branch'];
         }
 
-        return $defaultBranch;
+        if ($repository->isBitBucket()) {
```

Comment:
I think that correct name is Bitbucket and not BitBucket

Reply:
ops!! I'll fix..